File: 06_basic_hashing/01_counting_frequencies/solution.java
// https://www.geeksforgeeks.org/problems/frequency-of-array-elements-
class Solution {
    // Function to count the frequency of all elements from 1 to N in the array.
    public List<Integer> frequencyCount(int[] arr) {
        ArrayList<Integer> result = new ArrayList<>();
        
        for (int i = 0; i < arr.length; i++) {
            result.add(0);
        }
        
        for (int element: arr) {
            result.set(element-1, result.get(element-1) + 1);
        }
        
        return result;
    }
}

File: 06_basic_hashing/01_counting_frequencies/solution.py
# https://www.geeksforgeeks.org/problems/frequency-of-array-elements
class Solution:
    # Function to count the frequency of all elements from 1 to N in the array.
    def frequencyCount(self, arr):
        result = [0] * len(arr)
        
        for element in arr:
            result[element-1] = 1 + result[element-1]
            
        return result